--- a/llama.cpp/src/llama-model-loader.cpp
+++ b/llama.cpp/src/llama-model-loader.cpp
@@ -452,7 +452,7 @@
             struct GGUFMeta::ArrayInfo arr_info =
                 GGUFMeta::GKV<GGUFMeta::ArrayInfo>::get_kv(metadata, kid);
 
-            if (n != arr_info.length) {
+            if (arr_info.length > n) {
                 throw std::runtime_error(format("key %s has wrong array length; expected %u, got %u", key.c_str(), n, (uint32_t) arr_info.length));
             }
 
